### PR TITLE
Replace old metric settings with new ones

### LINKED ISSUE #610: Replace old metric settings with new v2 metric settings page
* Let's move metric settings out of the "Settings" page -- so there should no longer be a "Metric Configuration" link on the left side here:

* Instead, let's add an new "Configuration" link to the top toolbar (right after "Data" and before "Learn More" that directs to the new v2 metric page

* Once this is done, we can delete all the code for the old metric settings page

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Replace old metric settings with new ones
Added blue icon near definition label on definitions subpage to indicate that at least one definition was selected
Moved old metric config styles to corresponding places
Some cleanups

## Type of change
Type: Non-breaking refactor

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #610 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -224,7 +223,7 @@ const Menu: React.FC = () => {
               }}
             >
               Get Started
-              {/* Guidance: Metric Configuration Progress Toast */}
+              {/* Guidance: Metric MetricsConfiguration Progress Toast */}
```
Comment (mxosman) [inline publisher/src/components/Menu/Menu.tsx:226]: nit/q: I think the search and replace picked this up - did you mean to change?

File: publisher/src/components/Guidance/Guidance.tsx [hunk 2]
```diff
@@ -345,7 +345,7 @@ export const Guidance = observer(() => {
           )}
         </ContentContainer>
 
-        {/* Configure Metrics: Overview List */}
+        {/* Configure Metrics: MetricsOverview List */}
```
Comment (mxosman) [inline publisher/src/components/Guidance/Guidance.tsx:348]: nit/q: was this intentional?

File: publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx [hunk 3]
```diff
@@ -105,23 +129,50 @@ function MetricDefinitions() {
...
+                      dimensionDefinitionSettings[systemMetricKey][
+                        disaggregationKey
+                      ][key]
+                    ).reduce((acc, dimensionSetting) => {
+                      return { ...acc, ...dimensionSetting.settings };
+                    }, {} as { [settingKey: string]: Partial<MetricConfigurationSettings> });
+                    hasEnabledDefinition = !!Object.values(
+                      dimensionSettings
+                    ).find((setting) => setting.included === "Yes");
+                  }
```
Comment (mxosman) [inline publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx:151]: Just for my understanding - can you clarify what's happening here? I'm a bit lost.

File: publisher/src/components/Menu/Menu.tsx [hunk 4]
```diff
@@ -284,6 +283,17 @@ const Menu: React.FC = () => {
...
         </MenuItem>
 
+        {/* Metric Config */}
+        <MenuItem
+          onClick={() => {
+            navigate("metric-config");
+            handleCloseMobileMenu();
+          }}
+          active={pathWithoutAgency === "metric-config"}
+        >
```
Comment (mxosman) [inline publisher/src/components/Menu/Menu.tsx:293]: Jessica called this out when playing around with it - when you hit the `Configuration` link in the nav bar while inside of the Configuration page, it redirects to `/metric-config` without the params and nothing loads. Is there a simple way to reinit the page so when you click on the Configuration link while inside that page it takes the user back to the metric overview? Could use `useLocation` and see if the search params are empty and add that as a dependency to the effect that initializes the metric config - but I'm curious if you have other ideas.

**Review by mxosman — COMMENTED** [review]
Hi Ivan! Thank you so much for taking this on and for all of the extra clean up and reorganization you did along the way - I truly appreciate it. I left some questions & comments but nothing glaring stood out to me.

There is one weird thing I ran into that I'm having a tough time finding in another agency - so it may just be that agency and it's state in the DB, but let me know if you're able to repro/have an ideas what might be going on. 

CC Jail agency (in video) > Pre-adjudication Daily Population metric > Adjust the `Biological Sex` breakdown 

https://user-images.githubusercontent.com/59492998/236580913-e7d12946-4a35-4a5a-b465-22fd25bec870.mov

I'll give others a chance to review and play with this as well before we decide this is ready to launch.

**Comment by ivan-kishko** [comment]
> Hi Ivan! Thank you so much for taking this on and for all of the extra clean up and reorganization you did along the way - I truly appreciate it. I left some questions & comments but nothing glaring stood out to me.
> 
> There is one weird thing I ran into that I'm having a tough time finding in another agency - so it may just be that agency and it's state in the DB, but let me know if you're able to repro/have an ideas what might be going on.
> 
> CC Jail agency (in video) > Pre-adjudication Daily Population metric > Adjust the `Biological Sex` breakdown
> 
>  Screen.Recording.2023-05-05.at.5.13.06.PM.mov 
> I'll give others a chance to review and play with this as well before we decide this is ready to launch.

Hello @mxosman. I kinda can not understand what is the problem here apart from empty overview page. I am gonna assume that you are confused with why if you enable one of the biological sex dimensions it is available for upload and if you disable all the dimensions it is available for publishing - if it is the case, when you disable all dimensions, disaggregation becomes disabled as well and thus this disaggregation and its dimensions removed from publish availability calculation (since whole disaggregation is disabled) and if you enable at least one dimension but it has zero selected definitions then it is not available for publishing.

File: publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx (hunk 3 above)
Comment (ivan-kishko) [reply publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx:151]: this calculation is made to display a blue check icon indicator near dimension if it has at least one selected definition (which affects availability for publishing). I forgot to do that in metrc definitions page task, if you check figma you get what I am talking about

File: publisher/src/components/Guidance/Guidance.tsx (hunk 2 above)
Comment (ivan-kishko) [reply publisher/src/components/Guidance/Guidance.tsx:348]: no, I was refactofing file names, gonna fix, sorry

File: publisher/src/components/Menu/Menu.tsx (hunk 1 above)
Comment (ivan-kishko) [reply publisher/src/components/Menu/Menu.tsx:226]: no, I was refactofing file names, gonna fix, sorry

File: publisher/src/components/Menu/Menu.tsx (hunk 4 above)
Comment (ivan-kishko) [reply publisher/src/components/Menu/Menu.tsx:301]: [830438d](https://github.com/Recidiviz/justice-counts/pull/613/commits/830438d88ba059554f57072bef5c672a616e2200)

File: publisher/src/components/Guidance/Guidance.tsx (hunk 2 above)
Comment (ivan-kishko) [reply publisher/src/components/Guidance/Guidance.tsx]: [830438d](https://github.com/Recidiviz/justice-counts/pull/613/commits/830438d88ba059554f57072bef5c672a616e2200)

File: publisher/src/components/Menu/Menu.tsx (hunk 1 above)
Comment (ivan-kishko) [reply publisher/src/components/Menu/Menu.tsx]: [830438d](https://github.com/Recidiviz/justice-counts/pull/613/commits/830438d88ba059554f57072bef5c672a616e2200)

**Review by lilidworkin — APPROVED** [review]
> I am gonna assume that you are confused with why if you enable one of the biological sex dimensions it is available for upload and if you disable all the dimensions it is available for publishing - if it is the case, when you disable all dimensions, disaggregation becomes disabled as well and thus this disaggregation and its dimensions removed from publish availability calculation 

Yea I actually think @ivan-kishko is right, it does seem to be working as intended! @mxosman let us know if that makes sense, happy to explain further

The functionality all looks good to me so I will accept! But @mxosman definitely feel free to call out any improvements to the code or organization that you'd like to see.

**Comment by mxosman** [comment]
> I am gonna assume that you are confused with why if you enable one of the biological sex dimensions it is available for upload and if you disable all the dimensions it is available for publishing - if it is the case, when you disable all dimensions, disaggregation becomes disabled as well and thus this disaggregation and its dimensions removed from publish availability calculation (since whole disaggregation is disabled) and if you enable at least one dimension but it has zero selected definitions then it is not available for publishing.

> Yea I actually think @ivan-kishko is right, it does seem to be working as intended! @mxosman let us know if that makes sense, happy to explain further

The disabling all dimensions === disabled disaggregation behavior makes sense to me. What was throwing me off in the Pre-adjudication Daily Population metric in the CCJ agency, was when all biological sex dimensions are enabled, the status is "Available for data upload". If you **_only_** turn off Female Biological Sex, it becomes "Available for publishing" - turn it back on and it's "Available for data upload". But, I think I understand now that is due to Female Biological Sex having no set definitions. Feels like this might be confusing/unintuitive behavior - or it's just me who was confused! 

Thank you for the clarification.

File: publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx (hunk 3 above)
Comment (mxosman) [reply publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx:151]: Gotcha! Thank you for clarifying.

One nit: can we use a var for `dimensionDefinitionSettings[systemMetricKey][disaggregationKey][key]`

**Comment by lilidworkin** [comment]
> If you only turn off Female Biological Sex, it becomes "Available for publishing" - turn it back on and it's "Available for data upload". But, I think I understand now that is due to Female Biological Sex having no set definitions. 

Yea this threw me off for a second two @mxosman ! But not sure how else to address so I think it's ok for now.

File: publisher/src/components/Menu/Menu.tsx (hunk 4 above)
Comment (mxosman) [reply publisher/src/components/Menu/Menu.tsx:301]: Even better! Thank you!

**Comment by mxosman** [comment]
Haha - thanks for the sanity check -- I have no ideas on how to address either and agree not blocking at all! 

I just have one tiny nit, but nothing else from my end. All good to go!

**Review by mxosman — APPROVED** [review]

**Comment by lilidworkin** [comment]
@ivan-kishko I'm going to go ahead and merge this, thanks for your hard work on it!

**Pull request merged**